src/base/bittorrent/torrentimpl.cpp

```diff
@@ -282,8 +281,10 @@ TorrentImpl::TorrentImpl(Session *session, lt::session *nativeSession
         {
             const lt::file_index_t nativeIndex = m_torrentInfo.nativeIndexes().at(i);
             m_indexMap[nativeIndex] = i;
-            const QString filePath = Utils::Fs::toUniformPath(QString::fromStdString(fileStorage.file_path(nativeIndex)));
-            m_filePaths.append(filePath.endsWith(QB_EXT, Qt::CaseInsensitive) ? filePath.chopped(QB_EXT.size()) : filePath);
+            Path filePath {fileStorage.file_path(nativeIndex)};
+            if (filePath.hasExtension(QB_EXT))
```

Comment:
This causes performance regression during startup
![image](https://user-images.githubusercontent.com/34717789/153815381-c178fe82-5d09-49ac-84c2-d6301b07e6ef.png)

Reply:
> This causes performance regression during startup ![image](https://user-images.githubusercontent.com/34717789/153815587-6cf51733-f4f7-4df8-a348-1999796c9077.png)

#16444 should improve it.

Reply:
Is it necessary to use `QMimeDataBase::suffixForFileName` for extension? Maybe leave that to the user if they require a MIME base suffix, they can call it by themselves.